app/dao/fact_billing_dao.py

```diff
@@ -19,11 +24,155 @@
     NOTIFICATION_STATUS_TYPES_BILLABLE,
     NotificationHistory,
     EMAIL_TYPE,
-    NOTIFICATION_STATUS_TYPES_BILLABLE_FOR_LETTERS
+    NOTIFICATION_STATUS_TYPES_BILLABLE_FOR_LETTERS,
+    AnnualBilling,
+    Organisation,
 )
 from app.utils import get_london_midnight_in_utc
 
 
+def fetch_sms_free_allowance_remainder(start_date):
+    # ASSUMPTION: AnnualBilling has been populated for year.
+    billing_year = which_financial_year(start_date)
+    print(billing_year)
+    start_of_year = convert_utc_to_bst(financial_year_start(billing_year))
+    query = db.session.query(
+        FactBilling.service_id.label("service_id"),
+        AnnualBilling.free_sms_fragment_limit,
+        func.sum(FactBilling.billable_units * FactBilling.rate_multiplier).label('billable_units'),
+        func.greatest((AnnualBilling.free_sms_fragment_limit -
+                       func.sum(FactBilling.billable_units * FactBilling.rate_multiplier)
+                       ).cast(Integer), 0).label('sms_remainder')
+    ).filter(
+        FactBilling.service_id == Service.id,
+        FactBilling.service_id == AnnualBilling.service_id,
+        FactBilling.bst_date >= start_of_year,
+        FactBilling.bst_date < start_date,
+        FactBilling.notification_type == SMS_TYPE,
+        AnnualBilling.financial_year_start == billing_year,
+    ).group_by(
+        FactBilling.service_id,
+        AnnualBilling.free_sms_fragment_limit,
+    )
+    return query
+
+
+def fetch_sms_billing_for_all_services(start_date, end_date):
+
+    # ASSUMPTION: AnnualBilling has been populated for year.
+    billing_year = which_financial_year(start_date)
+    free_allowance_remainder = fetch_sms_free_allowance_remainder(start_date).subquery()
+    sms_billable_units = func.sum(FactBilling.billable_units * FactBilling.rate_multiplier)
+    sms_remainder = func.coalesce(free_allowance_remainder.c.sms_remainder, AnnualBilling.free_sms_fragment_limit)
+    chargeable_sms = case([(sms_remainder == 0, sms_billable_units),
+                           (sms_billable_units - sms_remainder <= 0, 0),
+                           (sms_billable_units - sms_remainder > 0,
+                            sms_billable_units - sms_remainder)], else_=0)
+    sms_cost = case([(sms_remainder == 0, sms_billable_units * FactBilling.rate),
+                     (sms_billable_units - sms_remainder <= 0, 0),
+                     (sms_billable_units - sms_remainder > 0, (sms_billable_units - sms_remainder) * FactBilling.rate)
+                     ], else_=0)
```

Comment:
i think this can just be `func.max((sms_billable_units - sms_remainder) * FactBilling.rate, 0)`